src/main/java/uk/gov/hmcts/darts/event/service/impl/EventHandlerBase.java

```diff
@@ -66,4 +95,107 @@ protected String buildKey(String type, String subType) {
         requireNonNull(type);
         return type + (isNull(subType) ? "" : subType);
     }
+
+    protected CourtroomCourthouseCourtcase getOrCreateCourtroomCourtHouseAndCases(DartsEvent dartsEvent) {
+        final var actualCourtRoomEntity = getCourtRoom(dartsEvent);
+
+        final var caseNumbers = dartsEvent.getCaseNumbers();
+        if (caseNumbers.size() > 1) {
+            log.warn(format(MULTIPLE_CASE_NUMBERS, dartsEvent.getEventId(), join(", ", caseNumbers)));
+        }
+        final var actualCourtHouse = actualCourtRoomEntity.getCourthouse();
+        final var actualEventCaseNumber = caseNumbers.get(0);
+        final var actualEventDate = dartsEvent.getDateTime();
+
+        var actualCourtCaseEntity = getCourtCaseEntityOrCreate(actualCourtHouse, actualEventCaseNumber);
+
+        boolean hearingIsNew = false;
+        HearingEntity actualHearingEntity;
+        var optionalHearingEntity = getHearingEntity(actualEventDate, actualCourtCaseEntity);
```

Comment:
i think this logic is wrong. i think we need to be checking what courtroom the hearing is in also, as we could have had 2 hearings for the same case in 1 day.
1 in courtroom 1, another in courtroom2.
Therefore we want to make sure that it is the correct hearing.

Here is another scenario:-
Lets say we have 1 hearing in the hearing table, that is for courtroom 1 and is only scheduled (hearingIsActual is false).
lets say, they chnged their mind, and it actually happened in courtroom 2, when we receive teh event, we will look for a hearing in courtrroom 2 and not find one, but before creating it, lets check that there is any hearings in this courthouse that have hearingIsActual==false, if we do find one, then we want to grab it, update it to be the correct courtroom and set the hearingIsActual=true